packages/std/src/traits.rs

```diff
@@ -61,12 +61,17 @@ pub trait Storage {
     /// The bound `start` is inclusive and `end` is exclusive.
     /// If `start` is lexicographically greater than or equal to `end`, an empty range is described, mo matter of the order.
     #[cfg(feature = "iterator")]
```

Comment:
No, we don't need to do that. What makes the feature non-additive right now is the situation when you implement the trait without the function (because you don't want to require the iterator feature) and something else then enables the feature. Then you suddenly break because your impl is missing a method.
Adding a default impl fixes that because your Storage impl now inherits the default impl.

or tl;dr: For a feature to be additive, it must only introduce non-breaking changes. Adding a trait method with a default impl is a non-breaking change.